### DISCUSSION TITLE
Failed, called error. code -32602 ,failed to decode textDocument/rename request: expected object at (root).textDocument from: textDocument/rename

### DISCUSSION THREAD

**Discussion by Hebe970217** [clangd/clangd#1048] [Q&A] [answered]
Request： 
`{
    "id": 7,
    "jsonrpc": "2.0",
    "method": "textDocument/rename",
    "params": {
        "newName": "T1",
        "position": {
            "character": 8,
            "line": 11
        },
        "textDocument": "file:///home/funning/workspace/workspace/recode/gg/TestApp/main.cpp"
    }
}`

Response：
`{"error":{"code":-32602,"message":"failed to decode textDocument/rename request: expected object at (root).textDocument"},"id":7,"jsonrpc":"2.0"}`

I don't have any ideas about this problem. Do anybody know what causes this problem?

**Comment by HighCommander4** [comment, marked as answer]
What client is this?

**Reply by HighCommander4** [reply]
Anyways, the issue at the protocol level is that `"textDocument"` is expected to be a `TextDocumentIdentifier` object (see [`TextDocumentPositionParams`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentPositionParams)), not a string, so this is a case of the client sending a malformed request.

**Reply by Hebe970217** [reply]
My create a new IDE，it's custom, protocol keywords is same from Kate(IDE for KDE)

**Reply by Hebe970217** [reply]
In order to better understand LSP protocol, I did not use the C + + SDK provided

**Reply by Hebe970217** [reply]
I solved the problem
Change` "textDocument": "file:///home/funning/workspace/workspace/recode/gg/TestApp/main.cpp" `
to `"textDocument": {"uri" : "file:///home/funning/workspace/workspace/recode/gg/TestApp/main.cpp" }`

is working narmal
